### PR TITLE
Adding new file-selector component

### PR DESCRIPTION
**Pull request by stulees**
- To be used in a widget settings form where a selection/input is required of a storage file/folder or custom url for a file
- Can optionally specify file type (image or video)
- Title and storage button labels applied via attributes
- `circle.yml` specifying `4.2.2` Node version
- e2e test coverage

### CODE REVIEW

**Comment by stulees** [comment]
@rodrigopavezi @donnapep Please review. FYI, when ready, I will ensure PR #217 gets merged first and this PR will be rebased before merging. Thanks

**Comment by stulees** [comment]
@donnapep Inline comments added to `getStorageName()` function of `dtv-file-selector.js`, thanks

**Comment by donnapep** [comment]
@stulees Nice. Thx! :+1:

**Comment by stulees** [comment]
@donnapep @rodrigopavezi Minor update, please review last commit, thanks

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 machine:
   node:
-   version: 0.12.2
```
Comment (rodrigopavezi) [inline circle.yml]: @stulees I guess it probably needs the new gcc also. cheers

**Comment by rodrigopavezi** [comment]
@stulees just a commend re the node version and cci script. Beside that, looks good. :+1:

File: circle.yml (hunk 1 above)
Comment (stulees) [inline circle.yml]: @rodrigopavezi Yeah it did, we'll be adding it to repos on a case by case basis I guess

**Pull request merged**